tools/bundle-open-api-tpl-files.ts

```diff
@@ -0,0 +1,262 @@
+import { fileURLToPath } from "node:url";
+import path from "node:path";
+import yargs from "yargs";
+import { hideBin } from "yargs/helpers";
+import fs from "fs-extra";
+import { globby, Options as GlobbyOptions } from "globby";
+import { RuntimeError } from "run-time-error";
+import prettier from "prettier";
+import { OpenAPIV3_1 } from "openapi-types";
+import { bundle, loadConfig, NormalizedProblem } from "@redocly/openapi-core";
+import { BundleResult } from "@redocly/openapi-core/lib/bundle";
+
+const TAG = "[tools/bundle-open-api-tpl-files.ts]";
+
+export interface IBundleOpenApiTplFilesV1Request {
+  readonly argv: string[];
+  readonly checkGitDiffDisabled: boolean;
+  readonly specTemplateGlobs: Readonly<Array<string>>;
+  readonly specTemplateIgnoreGlobs: string[];
+  readonly verbose: boolean;
+  readonly workingDirectory: Readonly<string>;
+}
+
+export interface IBundleOpenApiTplFilesV1Response {
+  readonly request: Readonly<IBundleOpenApiTplFilesV1Request>;
+  readonly specFilePaths: Readonly<Array<string>>;
+  readonly bundleProblems: Readonly<Record<string, NormalizedProblem[]>>;
+  readonly totalBundleProblemCount: number;
+  readonly timestamp: Readonly<string>;
+}
+
+const nodePath = path.resolve(process.argv[1]);
+const modulePath = path.resolve(fileURLToPath(import.meta.url));
+const isRunningDirectlyViaCLI = nodePath === modulePath;
+
+const main = async (argv: string[], env: NodeJS.ProcessEnv) => {
+  const req = await createRequest(argv, env);
+  await bundleOpenApiTplFiles(req);
+};
+
+if (isRunningDirectlyViaCLI) {
+  main(process.argv, process.env);
```

Comment:
Could you catch and pretty print any uncaught exceptions here?

Reply:
@outSH I stoppped doing that as a pattern because it appears that the error message gets propagated corerctly regardless. With that said, I agree it's probably better to be extra cautious because it is really annoying when an error gets swallowed! I'll updated it in a second!

![alt text](image.png)

Reply:
@outSH I updated the code, now the catch handler does log any uncaught exceptions before exiting with 1